internal/app/admin/http/cache_test.go

```diff
@@ -699,6 +501,42 @@ func TestAdminServer_CacheDumpHandler_WildcardSuffix_NotFound(t *testing.T) {
 	}
 }
 
+func verifyCacheOutput(t *testing.T, rr *httptest.ResponseRecorder, cdsFile string, ldsFile string) {
+	var actualResponse map[string]interface{}
+	err := json.Unmarshal(rr.Body.Bytes(), &actualResponse)
+	assert.NoError(t, err)
+
+	filecontentsCds, err := ioutil.ReadFile(cdsFile)
+	assert.NoError(t, err)
+	var expectedCdsResponse map[string]interface{}
+	err = json.Unmarshal(filecontentsCds, &expectedCdsResponse)
+	assert.NoError(t, err)
+
+	filecontentsLds, err := ioutil.ReadFile(ldsFile)
+	assert.NoError(t, err)
+	var expectedLdsResponse map[string]interface{}
+	err = json.Unmarshal(filecontentsLds, &expectedLdsResponse)
```

Comment:
I believe the original struct representing `expectedLdsResponse` is `marshallableResource`, defined [here](https://github.com/envoyproxy/xds-relay/blob/master/internal/app/admin/http/cache.go#L182). `jsonpb.UnmarshalString` expects a proto message, so it wouldn't work.

Reply:
sg..can we use the `marshallableResource` struct instead of interface?

Reply:
I had trouble when doing because the struct contains proto fields `types.Resource`. So, it will fail when attempting to unmarshal those fields using the json function.